### PR TITLE
[COOK-3845] Allow checks for single hosts, not just host groups

### PR DESCRIPTION
**Pull request by wpietri**
https://tickets.opscode.com/browse/COOK-3845

My chef setup isn't huge, so I have some boxes that are unique. So that I don't have to add spurious host groups, I have patched the Nagios server recipe to handle individual hosts as well as host groups.

### CODE REVIEW

File: templates/default/services.cfg.erb [hunk 1]
```diff
@@ -14,11 +14,19 @@ define service {
 # Services defined by data bags
 <% @services.each do |service| -%>
   <%# Do not define a service that has no hostgroup or a hostgroup that doesn't exist (including the "all" hostgroup) %>
-  <% unless service['hostgroup_name'].nil? || !(@hostgroups.include?(service['hostgroup_name']) || @search_hostgroups.include?(service['hostgroup_name']) || service['hostgroup_name'] == "all") -%>
+<%
+  valid_hostgroup = !service['hostgroup_name'].nil? && (@hostgroups.include?(service['hostgroup_name']) || @search_hostgroups.include?(service['hostgroup_name']) || service['hostgroup_name'] == "all")
+  valid_hostname = !service['host_name'].nil? # does this need more checks?
+%>
```
Comment (btm) [inline templates/default/services.cfg.erb:20]: It's not immediately clear to me looking at the code what is going on here. There's a comment from before, so I'm wondering if there should be an addition to the comment for this section?

Comment (wpietri) [inline templates/default/services.cfg.erb:20]: I found the code a bit opaque myself, and I tried to make it clearer by naming the existing code 'valid_hostgroup', and then introduced a 'valid_hostname' as parallel. Normally I'd push domain logic like this out of the templates, but I couldn't see where to do this here, and I didn't want to be too ambitious for my first PR.

What if we changed the comment to something like: "Only define services that are for single hosts or existing hostgroups"?

**Comment by sbotman** [comment]
Hi William,

I think it counts for this one also. With the new providers this becomes obsolete code in my understanding. Are you oke if we close this one, or do we need to rebase and update the PR?

Gr Sander.

**Comment by wpietri** [comment]
Feel free to close. I'm running on my own forked cookbook for now, and don't have time (or inclination) to upgrade to whatever new thing is going on. Thanks for asking.

**Pull request closed without merging**